docs/building.md

```diff
@@ -215,6 +215,60 @@ Currently Loaded Modules:
   3) lsf-tools/2.0   6) cuda/10.1.243           9) spectrum-mpi/10.3.1.2-20200121
 ```
 
+### LANL Darwin (Heterogeneous)
+
+#### Allocate Node
+
+Darwin is a heterogeneous cluster, giving LANL developers easy access to a
+wide variety of architectures. Therefore, before you do anything else, you
+should allocate a node in the partition you intend to work in. Currently any
+partition with either Haswell or newer x86-64 nodes (e.g. `general`,
+`skylake-gold`, `skylake-platinum`), or the `power9` partition will do.
+
+E.g.
+```bash
+$ salloc -p power9
+```
+
+#### Set-Up Environment (Optional, but Still Recommended, for Non-CUDA Builds)
+
+You can import all tools you need to start building with by sourcing the
+project `.bashrc`:
+
+```bash
+$ source /projects/parthenon-int/parthenon-project/.bashrc
```

Comment:
it might be nice to store this somewhere on the parthenon repo (or similar), maybe in the wiki? I'd like the environment code and the cmake tied together somehow.

Reply:
It's stored in a separate, internal git repo that manages the project space layout.

So, the script is intentionally very brief and it doesn't actually load all that much into the user's environment - just cmake, ninja, and a recent git build, as well as setting `MACHINE_CFG`. Because it's sparse like this, the goal is you should always be able to use the latest .bashrc with any version of Parthenon. Instead, coupling between versions of Parthenon and dependencies is achieved by encoding the view "date" in the CMake file here.

One reason I wanted to put it in a known directory on darwin is that users can stick that exact string into their profiles and not have to think about it.

Thoughts?

See: https://xcp-gitlab.lanl.gov/eap-oss/parthenon-project

Currently working on a README for it.

Reply:
> One reason I wanted to put it in a known directory on darwin is that users can stick that exact string into their profiles and not have to think about it.

I agree with this strongly.

> So, the script is intentionally very brief and it doesn't actually load all that much into the user's environment

Oh yeah, I see. It's very simple. I guess since there's basically nothing in the script, it's probably fine not to have them together in the same repo. Maybe if the user doesn't want to use the bashrc, we could provide documentation somewhere for what it's doing?

Also, why Ninja and not make?

Reply:
I'm pretty sure Ninja is faster.

Reply:
But I hate change. :P

Reply:
Yeah, Ninja completes a clean CUDA build ~70% faster on power9, in my experience.

Reply:
It's essentially CMake's fault, unfortunately - its Make generator won't overlap compilation of source files from dependent targets. So it builds all of Kokkos, links it, then builds all of parthenon, links it, and then builds the examples and tests. The Ninja generator doesn't suffer from this.